packages/eslint-plugin/tests/rules/naming-convention.test.ts

```diff
@@ -2224,5 +2224,17 @@ ruleTester.run('naming-convention', rule, {
       ],
       errors: Array(13).fill({ messageId: 'doesNotMatchFormat' }),
     },
+    {
+      code: `
+        type Foo = {
+          'foo     Bar': string;
```

Comment:
A few missing test cases:

* `'': ...`
* `'0': ...` _(you never know with numbers...)_
* `'foo': ...` _(unless I'm missing this already existing?
* `'foo-bar': ...`

Reply:
It seems like quoting makes some crazy naming possible in general. I didn't include it here but tested with something like `'!!@#$wow': string;` on typescript's playground

Reply:
I'd like to put `'foo': string;` but the CI lint failed. Anyway, that 'foo' passes the test because the default setting is camelCase(so it is not included in the error count)

Reply:
We typically get around that with:

```ts
// eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
```

Reply:
instead you can just use the `noFormat` template tag on this test to bypass the formatting.

Reply:
hahaha documenting can be challenging sometime